src/arkode/arkode_splittingstep.c

```diff
@@ -0,0 +1,635 @@
+  ---------------------------------------------------------------*/
+static void splittingStep_PrintMem(const ARKodeMem ark_mem, FILE* const outfile)
+{
+  ARKodeSplittingStepMem step_mem = NULL;
+  const int retval = splittingStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return; }
+
+  /* output integer quantities */
+  fprintf(outfile, "SplittingStep: partitions = %i\n", step_mem->partitions);
+  fprintf(outfile, "SplittingStep: order = %i\n", step_mem->order);
+
+  /* output long integer quantities */
+  for (int k = 0; k < step_mem->partitions; k++)
+  {
+    fprintf(outfile, "SplittingStep: partition %i: n_stepper_evolves = %li\n",
+            k, step_mem->n_stepper_evolves[k]);
+  }
+
+  /* output sunrealtype quantities */
+  fprintf(outfile, "SplittingStep: Coefficients:\n");
+  SplittingStepCoefficients_Write(step_mem->coefficients, outfile);
+}
+
+/*---------------------------------------------------------------
+  Specifies the method order
+  ---------------------------------------------------------------*/
+static int splittingStep_SetOrder(const ARKodeMem ark_mem, const int order)
+{
+  ARKodeSplittingStepMem step_mem = NULL;
+  const int retval = splittingStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return retval; }
+
+  /* set user-provided value, or default, depending on argument */
+  step_mem->order = order <= 0 ? 1 : order;
+
+  SplittingStepCoefficients_Free(step_mem->coefficients);
+  step_mem->coefficients = NULL;
+
+  return ARK_SUCCESS;
+}
+
+/*---------------------------------------------------------------
+  Resets all SplittingStep optional inputs to their default
+  values. Does not change problem-defining function pointers or
+  user_data pointer.
+  ---------------------------------------------------------------*/
+static int splittingStep_SetDefaults(const ARKodeMem ark_mem)
+{
+  ARKodeSplittingStepMem step_mem = NULL;
+  int retval = splittingStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return retval; }
+
+  retval = splittingStep_SetOrder(ark_mem, 0);
+  if (retval != ARK_SUCCESS) { return retval; }
+
+  /* TODO(SBR): This may cause issues if a user calls ARKodeSetDefaults. This
+   * issues affects other ARKODE steppers as well */
+  ARKodeSetInterpolantType(ark_mem, ARK_INTERP_LAGRANGE);
+
+  return ARK_SUCCESS;
+}
+
+/*---------------------------------------------------------------
+  Creates the SplittingStep integrator
+  ---------------------------------------------------------------*/
+void* SplittingStepCreate(SUNStepper* const steppers, const int partitions,
+                          const sunrealtype t0, const N_Vector y0,
+                          const SUNContext sunctx)
+{
+  if (steppers == NULL)
+  {
+    arkProcessError(NULL, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "steppers = NULL illegal.");
+    return NULL;
+  }
+
+  for (int i = 0; i < partitions; i++)
```

Comment:
I went with the extra validation route. Supporting 1 partition will require special cases and testing for an edge case that is only really useful for debugging. It can be added later if we find a strong need.